src/test/java/com/connexta/search/common/SearchITests.java

```diff
@@ -7,430 +7,543 @@
+ */
+@SpringBootTest(webEnvironment = WebEnvironment.RANDOM_PORT)
+@AutoConfigureWebTestClient
 public class SearchITests {
 
-  private static final int SOLR_PORT = 8983;
+  @TestConfiguration
+  static class Config {
 
-  @Container
-  public static final GenericContainer solrContainer =
-      new GenericContainer("solr:8.1.1")
-          .withCommand("solr-create -c " + SOLR_COLLECTION)
-          .withExposedPorts(SOLR_PORT)
-          .waitingFor(Wait.forHttp("/solr/" + SOLR_COLLECTION + "/admin/ping"));
+    @Bean
+    public URL solrUrl() throws MalformedURLException {
+      return new URL("http", "localhost", 9983, "/solr");
+    }
+  }
 
-  private static final String INDEX_ENDPOINT_BASE_URL = "/index/";
+  @Inject
+  private SolrClient solrClient; // Injected to empty solr for each test. Not tested directly.
 
-  @NotNull
-  private static final Map<String, String> getSampleDatatHavingAllAttributes =
-      Map.of(
-          ID_ATTRIBUTE_NAME,
-          "00067360b70e4acfab561fe593ad3f7b",
-          CONTENTS_ATTRIBUTE_NAME,
-          "Winterfell",
-          MEDIA_TYPE_ATTRIBUTE_NAME,
-          "application/json");
-
-  private static final String allAttributesQuery =
-      QUERY_TERMS.stream()
-          .map(
-              term -> String.format("%s = '%s'", term, getSampleDatatHavingAllAttributes.get(term)))
-          .collect(Collectors.joining(" AND "));
-
-  @Inject private QueryManager queryManager;
-  @Inject private IndexManager indexManager;
-  @Inject private TestRestTemplate restTemplate;
-  @Inject private SolrClient solrClient;
-
-  @Value("${endpointUrl.productRetrieve}")
-  private String productRetrieveEndpoint;
+  @Inject private WebTestClient webTestClient;
 
   @Value("${endpoints.index.version}")
   private String indexApiVersion;
 
+  private static MockWebServer storeMockWebServer;
+
   @BeforeEach
   public void beforeEach() throws IOException, SolrServerException {
+    storeMockWebServer = new MockWebServer();
+    storeMockWebServer.start();
+
     // TODO shouldn't need to clear solr every time
     solrClient.deleteByQuery(SOLR_COLLECTION, "*");
     solrClient.commit(SOLR_COLLECTION);
   }
 
-  @Test
-  public void testContextLoads() {}
-
-  @ParameterizedTest
-  @ValueSource(strings = {"", " ", "text"})
-  public void testStoringValidCst(String contents) throws Exception {
-    // given
-    final String indexEndpointUrl = INDEX_ENDPOINT_BASE_URL + "00067360b70e4acfab561fe593ad3f7a";
-
-    // when indexing a product
-    ResponseEntity<String> response =
-        restTemplate.exchange(
-            indexEndpointUrl,
-            HttpMethod.PUT,
-            createIndexRequest("{ \"ext.extracted.text\" : \"" + contents + "\" }"),
-            String.class);
-
-    // then
-    assertThat(response.getStatusCode(), is(HttpStatus.OK));
+  @AfterEach
+  public void afterEach() throws IOException {
+    storeMockWebServer.shutdown();
```

Comment:
❓ Why restart the web server after each test?

Reply:
The requests stack up for each test method if you don't restart after every test. `getRequestCount()` (which is used in `testIndexWhenDatasetHasAlreadyBeenIndexed` to verify that the irmUri was only used once) returns the full amount of requests since the server was started.